File: src/Asm.BankPlus.Models/Transaction.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Asm.BankPlus.Models
{
    public class Transaction
    {
        public Guid Id { get; set; } = new Guid();
        public Guid? Reference { get; set; }
        public Guid AccountId { get; set; }
        public decimal Amount { get; set; }
        public string Description { get; set; }
        public DateTime TransactionTime { get; set; }

        public TransactionType TransactionType { get; set; }

        public IEnumerable<TransactionTag> Tags { get; set; }

        public object ExtraInfo { get; set; }
    }
}

File: src/Asm.BankPlus.Web/MooBankApp/src/models/Transaction.ts
import { TransactionTag } from "./TransactionTag";

export enum TransactionType {
    Credit = 1,
    Debit = 2,
    RecurringCredit = 3,
    RecurringDebit = 4,
    BalanceAdjustment = 5,
}

export interface Transaction {
    id: string;
    reference: string;
    accountId: string;
    amount: number;
    description: string;
    transactionTime: string;
    transactionType: TransactionType;
    tags: TransactionTag[];
    extraInfo: any;
}